incubator/vault/templates/deployment.yaml

```diff
@@ -30,13 +36,21 @@ spec:
           tcpSocket:
             port: {{ .Values.service.port }}
         readinessProbe:
-          tcpSocket:
-            port: {{ .Values.service.port }}
+            httpGet:
+              path: /v1/sys/health
+              port: {{ .Values.service.port }}
         securityContext:
           readOnlyRootFilesystem: true
           capabilities:
             add:
             - IPC_LOCK
+        env:
+          - name: VAULT_CLUSTER_ADDR
+            value: "http://$(POD_IP):8201"
+          - name: POD_IP
```

Comment:
`POD_IP` must be declared before `VAULT_CLUSTER_ADDR`, otherwise it doesn't interpolate. I just spent an hour wondering why it wasn't working...